### PR TITLE
Migration to bem-xjst 8

### PR DESCRIPTION
**Pull request by d3m1d0v**

### CODE REVIEW

File: .enb/make.js [hunk 1]
```diff
@@ -107,7 +107,12 @@ function createTmplTestsNodes(config) {
             },
             bemhtml : {
                 tech : 'enb-bemxjst/techs/bemhtml',
-                options : { sourceSuffixes : ['bemhtml', 'bemhtml.js'] }
+                options : {
+                    sourceSuffixes : ['bemhtml', 'bemhtml.js'],
+                    engineOptions : {
+                        xhtml : true
```
Comment (miripiruni) [inline .enb/make.js]: In v8.x default value is `true`.

**Comment by tadatuta** [comment]
let's also:
1. rename all the `*.bemhtml` files to `*.bemhtml.js`
2. in https://github.com/bem-contrib/bem-suggest/blob/master/blocks/suggest/_has-dataprovider/suggest_has-dataprovider.bemhtml#L2 replace 

```js
return this.ctx.dataprovider;
```
with
```js
var prevJs = applyNext();
return prevJs && prevJs !== true ? this.extend(prevJs, this.ctx.dataprovider) : this.ctx.dataprovider;
```

**Comment by belozer** [comment]
@tadatuta maybe better use `addJs()`? This is migration to 8.x

```js
block('suggest').match(function() { return this.mods['has-dataprovider']; }).addJs()(function() {
    return this.ctx.dataprovider;
})
```

**Comment by belozer** [comment]
And replace `match` to `mod`

```js
block('suggest').mod('has-dataprovider').addJs()(function() {
    return this.ctx.dataprovider;
})
```

**Comment by tadatuta** [comment]
@belozer 
> maybe better use addJs()? This is migration to 8.x

let's keep it backward compatible with bem-xjst < 8 as it's so easy for us.

**Comment by tadatuta** [comment]
@d3m1d0v thanks!

**Pull request merged**